plugins/node/opentelemetry-instrumentation-azure-functions/README.md

```diff
@@ -0,0 +1,58 @@
+# OpenTelemetry Azure Functions Instrumentation for Node.js
+
+[![NPM Published Version][npm-img]][npm-url]
+[![Apache License][license-image]][license-image]
+
+This module provides automatic instrumentation for the [`@azure/functions`](https://learn.microsoft.com/azure/azure-functions/functions-reference-node?pivots=nodejs-model-v4) module, which may be loaded using the [`@opentelemetry/sdk-trace-node`](https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-sdk-trace-node) package and is included in the [`@opentelemetry/auto-instrumentations-node`](https://www.npmjs.com/package/@opentelemetry/auto-instrumentations-node) bundle.
```

Comment:
Describing that the module does would be great here, looks like you are doing only context propagation right now, I wonder if it makes sense to have this package as part of the [propagators](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/propagators) in this repo instead. @open-telemetry/javascript-approvers any suggestions here?

Reply:
Interesting, I see a lot fewer examples for propagators but I will look into that.

For the record, this isn't all the package will ever do. Unfortunately I'm blocked from adding some more functionality while I wait for some Azure rollouts. For example, here's a small commit that illustrates how we also want to emit logs from here: https://github.com/ejizba/opentelemetry-js-contrib/commit/21b8e1d85e6b0f785bebb1a4accabb8b6519d023

Reply:
We discussed this in today's SIG meeting, and it should be fine to have this and other future functionalities here as an instrumentation package.

Reply:
Our changes finished rolling out in Azure, so I was able to finally add the rest of the main functionality we wanted.